packages/x-internals/src/warning/slotWarning.ts

```diff
@@ -0,0 +1,14 @@
+import { warnOnce } from './warning';
+
+export function checkSlot(slotName: string, slot: Function) {
+  if (process.env.NODE_ENV !== 'production') {
+    if (slot.name !== undefined && slot.name === slotName) {
```

Comment:
The downside is a false negative in this case:

```tsx
function toolbar() {
  return <div />;
}
<DataGrid slots={{ toolbar: toolbar }} />
```

Reply:
For the 1st one, we could improve the error message to tell to people to just rename their function (if we think it's common enough).

For the 2nd one, I would love to have a way to detect it, but if we don't I guess it's better to do a PR that covers realisticaly 99.9% of the use cases than not doing it.

Reply:
@cherniavskii does your assertion work with this kind of synthax? https://github.com/mui/mui-x/issues/8322#issuecomment-1744822522

```ts
slots={{
  textField: renderDateInputTextField(false),
}}
```

I think they are more common than the edge cases presented above.

Reply:
Thanks for this example, I wasn't aware of it 👍🏻

The expected behavior is to warn in that case, right?
If so, this is another example of false negative – the warning won't appear, because the slot function name is empty in that case.

I've updated the test suite to include more use cases. I'll look what we can do to better detect issues 👍🏻

Reply:
> The expected behavior is to warn in that case, right?

Yes
People could use something like `lodash/memoize` to create `renderDateInputTextField` but it's such a bad practice :grimacing: